spring-integration-mongodb/src/test/java/org/springframework/integration/mongodb/inbound/MongoDbMessageSourceTests.java

```diff
@@ -163,20 +164,38 @@ public void validateSuccessfullSubObjectQueryWithSinigleElementIfOneInList() thr
 	@MongoDbAvailable
 	public void validateSuccessfullQueryWithMultipleElements() throws Exception {
 
+		@SuppressWarnings("unchecked") List<Person> persons = queryMultipleElements(new LiteralExpression("{'address.state' : 'PA'}"));
+		assertEquals(3, persons.size());
+	}
+	@Test
+	@MongoDbAvailable
+	public void validateSuccessfullStringQueryExpressionWithMultipleElements() throws Exception {
+
+		@SuppressWarnings("unchecked") List<Person> persons = queryMultipleElements(new SpelExpressionParser()
```

Comment:
We don't need this `@SuppressWarnings("unchecked") ` here, but at the same time we need it in the `queryMultipleElements()` for the `List<Person>` cast:
```
MongoDbMessageSourceTests.java:198: warning: [unchecked] unchecked cast
                return (List<Person>) messageSource.receive().getPayload();

```